src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.cs

```diff
@@ -444,5 +444,635 @@ public void M(int x)
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
+
+    Jump if False (Regular) to Block[B3]
+        IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'a')
+
+    Next (Regular) Block[B2]
+Block[B2] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'b')
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
+
+    Next (Regular) Block[B4]
+Block[B3] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c')
+          Value: 
+            IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: System.String) (Syntax: 'c')
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'b = $""{(a ? ... c),20:D3}"";')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.String) (Syntax: 'b = $""{(a ? ...  c),20:D3}""')
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'b')
+              Right: 
+                IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String) (Syntax: '$""{(a ? b : c),20:D3}""')
+                  Parts(1):
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null) (Syntax: '{(a ? b : c),20:D3}')
+                        Expression: 
+                          IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'a ? b : c')
+                        Alignment: 
+                          ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 20) (Syntax: '20')
+                        FormatString: 
+                          ILiteralOperation (OperationKind.Literal, Type: System.String, Constant: ""D3"") (Syntax: ':D3')
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void InterpolatedStringExpression_NestedInterpolation_Flow()
+        {
+            string source = @"
+using System;
+
+internal class Class
+{
+    public void M(string a, int? b, int c)
+    /*<bind>*/{
+        a = $""{M2($""{b ?? c}"")}"";
```

Comment:
>M2($""{b ?? c}"") [](start = 16, length = 17)

Is there anything interesting about an invocation in this position? Please remove the test if not.

Reply:
Yes, there is a nested interpolation operation with a flow capture, and I chose to do so via an invocation in the between the outer and inner interpolations.

---
In reply to: [185917899](https://github.com/dotnet/roslyn/pull/26545#discussion_r185917899) [](ancestors = 185917899)

Reply:
> Yes, there is a nested interpolation operation with a flow capture, and I chose to do so via an invocation in the between the outer and inner interpolations.

What is unique about that? There are millions way to have variations in flow graph. Is there some kind of unique interaction between string interpolation and invocation inside it?

---
In reply to: [185918814](https://github.com/dotnet/roslyn/pull/26545#discussion_r185918814) [](ancestors = 185918814,185917899)

Reply:
I wanted to cover rewrite of a new interpolation string while we are already rewriting another interpolation string. I happened to choose an invocation in between but if you have better test to suggest, please do let me know.

---
In reply to: [185919670](https://github.com/dotnet/roslyn/pull/26545#discussion_r185919670) [](ancestors = 185919670,185918814,185917899)

Reply:
> I wanted to cover rewrite of a new interpolation string while we are already rewriting another interpolation string.

I don't believe this is interesting? What am I missing?

---
In reply to: [185921886](https://github.com/dotnet/roslyn/pull/26545#discussion_r185921886) [](ancestors = 185921886,185919670,185918814,185917899)

Reply:
I mean I don't believe it is interesting to have an invocation in between. It is interesting to test interpolation in another construct and an interpolation within an interpolation sounds like something that can cover that.

---
In reply to: [185923180](https://github.com/dotnet/roslyn/pull/26545#discussion_r185923180) [](ancestors = 185923180,185921886,185919670,185918814,185917899)

Reply:
It is interesting because there is conditional flow deeply nested in an interpolation. It might not be interesting knowing the implementation details of how the CFG builder works, but is indeed a unique user scenario of code with conditional flow.

---
In reply to: [185923180](https://github.com/dotnet/roslyn/pull/26545#discussion_r185923180) [](ancestors = 185923180,185921886,185919670,185918814,185917899)